src/stellar/soroban/soroban.ts

```diff
@@ -276,4 +235,222 @@ export class Soroban {
+
+        const messages = [
+          ...formattedStderr,
+          ...this.formatContractDeploymentMessage(formattedStdout),
+        ];
+
+        resolve(messages);
+      });
+    });
+  }
+
+  private formatContractDeploymentMessage(
+    messages: OutputMessage[],
+  ): OutputMessage[] {
+    const constractMessage = messages
+      .filter((m) => m.text.length > 0)
+      .find((message) => message.text.startsWith("C"));
+
+    if (!constractMessage) {
+      return messages;
+    }
+
+    const contractAddress = constractMessage.text;
+
+    return [
+      {
+        type: "text",
+        text: `Contract deployed successfully at address: ${contractAddress}`,
+      },
+    ];
+  }
+
+  private async checkContractConstructorArgs(
+    params: DeployContractArgs,
+  ): Promise<void> {
+    const { result } = await this.resolveContractArgs(params.wasmPath);
+
+    if (result && !params.constructorArgs) {
+      throw new Error(
+        `⚠️ Contract has a constructor but no arguments were provided`,
+      );
+    }
+  }
+
+  private async resolveContractArgs(
+    contractPath: string,
+  ): Promise<{ result: boolean; args: ConstructorArg[] }> {
+    const packageName = this.resolvePacakgeName(contractPath);
+    const contractArgs = this.readContractArgs(contractPath, packageName);
+
+    if (contractArgs.length > 0) {
+      return { result: true, args: contractArgs };
+    }
+
+    return { result: false, args: [] };
+  }
+
+  private readContractArgs(
+    contractPath: string,
+    packageName: string,
+    currentPath?: string,
+  ): ConstructorArg[] {
+    const srcDir =
+      currentPath || this.resolvePath(contractPath, "../../../../");
+    const entries = this.readDir(srcDir);
+    const packageNameInKebabCase = this.fromSnakeCaseToKebabCase(packageName);
+
+    for (const entry of entries) {
+      const fullPath = this.resolvePath(srcDir, entry);
+      const constructorArgs = this.extractConstructorArgsFromFile(
+        fullPath,
+        entry,
+      );
+
+      if (constructorArgs.length > 0) {
+        return constructorArgs;
+      }
+
+      if (this.shouldSearchDirectory(fullPath, entry, packageNameInKebabCase)) {
+        const nestedArgs = this.readContractArgs(
+          contractPath,
+          packageName,
+          fullPath,
+        );
+        if (nestedArgs.length > 0) {
+          return nestedArgs;
+        }
+      }
+    }
+
+    return [];
+  }
+
+  private extractConstructorArgsFromFile(
+    filePath: string,
+    fileName: string,
+  ): ConstructorArg[] {
+    if (!fileName.endsWith(".rs")) {
+      return [];
+    }
+
+    const content = this.readFile<string>(filePath);
+    const constructorMatch = content.match(/pub fn __constructor\(([^)]+)\)/);
+    if (!constructorMatch) {
+      return [];
+    }
+
+    return this.parseConstructorArgs(constructorMatch[1]);
+  }
+
+  private parseConstructorArgs(argsString: string): ConstructorArg[] {
```

Comment:
I think it's unnecessary to be redundant here. By typing this parameter as a string, we already know what type of value will be.
```suggestion
  private parseConstructorArgs(args: string): ConstructorArg[] {
```

Reply:
Actually, I think this should be just arg in singular because it is not an array of strings, right?
```suggestion
  private parseConstructorArgs(arg: string): ConstructorArg[] {
```

Reply:
I think we should keep this in plural, since even if it's a single string, it can contain multiple arguments. For example: `"(env: Env, admin: Address, age: i128, random_contract: Address)"`